// wp-timber-cli/lib/createSubclass.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _fs = require('fs');

var _fs2 = _interopRequireDefault(_fs);

var _mkdirp = require('mkdirp');

var _mkdirp2 = _interopRequireDefault(_mkdirp);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var createSubclass = function createSubclass(args) {
  var type = args[0];
  var name = args[1];

  if (selectClass(type)) {
    _mkdirp2.default.sync('./lib');
    _fs2.default.writeFileSync('./lib/' + name + '.php', buildPHPTemplate(type, name));
    console.log('Subclass Created!');
  } else {
    console.log('Please use a valid base class (post, term, menu, menuitem, user)');
  }
};

function buildPHPTemplate(type, name) {
  var baseClass = selectClass(type);

  return '<?php \n' + '\n' + 'Class ' + name + ' extends ' + baseClass + ' { \n' + '\t// Add methods and properties here \n' + '} \n';
  '?>';
}

function selectClass(type) {
  var classes = {
    site: 'TimberSite',
    post: 'TimberPost',
    term: 'TimberTerm',
    menu: 'TimberMenu',
    menuitem: 'TimberMenuItem',
    user: 'TimberUser'
  };
  if (classes[type]) {
    return classes[type];
  } else {
    return false;
  }
}

exports.default = createSubclass;
